src/ray/gcs/redis_context.cc

```diff
@@ -550,26 +541,14 @@ std::unique_ptr<CallbackReply> RedisContext::RunArgvSync(
   return callback_reply;
 }
 
-Status RedisContext::RunArgvAsync(const std::vector<std::string> &args,
-                                  const RedisCallback &redis_callback) {
+void RedisContext::RunArgvAsync(std::vector<std::string> args,
+                                RedisCallback redis_callback) {
   RAY_CHECK(redis_async_context_);
-  // Build the arguments.
-  std::vector<const char *> argv;
-  std::vector<size_t> argc;
-  for (size_t i = 0; i < args.size(); ++i) {
-    argv.push_back(args[i].data());
-    argc.push_back(args[i].size());
-  }
-  int64_t callback_index =
-      RedisCallbackManager::instance().AddCallback(redis_callback, io_service_);
-  // Run the Redis command.
-  Status status = redis_async_context_->RedisAsyncCommandArgv(
-      reinterpret_cast<redisCallbackFn *>(&GlobalRedisCallback),
-      reinterpret_cast<void *>(callback_index),
-      args.size(),
-      argv.data(),
-      argc.data());
-  return status;
+  auto request_context = new RedisRequestContext(io_service_,
```

Comment:
can we create a shared_ptr for request_context and use std::enable_shared_from_this for RedisRequestContext?